### PR TITLE
fix method regex

### PR DESCRIPTION
**Pull request by yuval-k**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2395

### CODE REVIEW

File: changelog/v1.3.5/envoy-deprecated-fields.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+changelog:
+- type: FIX
+  issueLink: https://github.com/solo-io/gloo/issues/2395
+  description: >
+    Use envoy's SafeRegex in method matching.
```
Comment (kdorosh) [inline changelog/v1.3.5/envoy-deprecated-fields.yaml]: mention that we are switching because the older regex match is now deprecated by envoy

Comment (yuval-k) [reply changelog/v1.3.5/envoy-deprecated-fields.yaml]: done

**Comment by kdorosh** [comment]
/kick e2e test flake introduced? https://console.cloud.google.com/cloud-build/builds/87700a43-5f2a-4ebe-8efe-6aed2d67197a;step=11?project=solo-public

```
Happy path in memory [It] should not crash multiple methods 
/workspace/gloo/test/e2e/happypath_test.go:112

  Timed out after 30.001s.
  Expected
      <*url.Error | 0xc002121680>: {
          Op: "Post",
          URL: "http://localhost:11108/1",
          Err: {
              Op: "dial",
              Net: "tcp",
              Source: nil,
              Addr: {
                  IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 127, 0, 0, 1],
                  Port: 11108,
                  Zone: "",
              },
              Err: {Syscall: "connect", Err: 0x6f},
          },
      }
  to be nil
```

**Pull request merged**